### PR TITLE
PHPC-1006: Do not modify Persistable::bsonSerialize() retval

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-1006

### CODE REVIEW

File: src/bson-encode.c [hunk 1]
```diff
@@ -556,6 +566,10 @@ void php_phongo_zval_to_bson(zval *data, php_phongo_bson_flags_t flags, bson_t *
 				goto cleanup;
 			}
 
+			if (skip_odm_field && !strcmp(string_key, PHONGO_ODM_FIELD_NAME)) {
```
Comment (bjori) [inline src/bson-encode.c]: Its confusing that string_key is both a char* and a zend_string.
but that might just be the context given github, which was very badly broken :|

**Comment by bjori** [comment]
I've no idea why this is failing on Travis but this lgtm

File: src/bson-encode.c (hunk 1 above)
Comment (jmikola) [reply src/bson-encode.c]: This diff line is for the PHP 5.x block, where `string_key` is the `char*` populated by `zend_hash_get_current_key_ex()`. In the preceding PHP 7.x block, `ZEND_HASH_FOREACH_KEY_VAL()` assigns the `zend_string`.

**Comment by jmikola** [comment]
> I've no idea why this is failing on Travis but this lgtm

Fixed in https://github.com/mongodb/mongo-php-driver/pull/644

**Pull request merged**